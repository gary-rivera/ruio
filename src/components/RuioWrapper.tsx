import React, { ReactNode } from 'react'
import { RuioContextProvider } from '@root/context/RuioContextProvider'
import ControlPanel from './ControlPanel'

// Single point of entry to ease the process of handling context throughout the module
const RuioWrapper: React.FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <RuioContextProvider>
      {/* <RuioLogo /> */}
      {/* <UtilityIcon /> */}
      <ControlPanel />
      {children}
    </RuioContextProvider>
  )
}

export default RuioWrapper
