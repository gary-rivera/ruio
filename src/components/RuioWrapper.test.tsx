import React from 'react'
import { render } from '@testing-library/react'
import RuioWrapper from './RuioWrapper'

describe('RuioWrapper Smoke Tests', () => {
  it('should render without crashing', () => {
    const { container } = render(
      <RuioWrapper>
        <div>Test Content</div>
      </RuioWrapper>,
    )
    expect(container).toBeInTheDocument()
  })

  it('should render child components', () => {
    const { getByText } = render(
      <RuioWrapper>
        <div>Child Component</div>
      </RuioWrapper>,
    )

    expect(getByText('Child Component')).toBeInTheDocument()
  })
})

describe('RuioWrapper Snapshot Tests', () => {
  it('should match the snapshot with default children', () => {
    const { asFragment } = render(
      <RuioWrapper>
        <div>Snapshot Child</div>
      </RuioWrapper>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('RuioWrapper Edge Cases', () => {
  it('should always render UtilityIcon and ControlPanel', () => {
    const { getByTestId } = render(
      <RuioWrapper>
        <div>Some Child</div>
      </RuioWrapper>,
    )

    expect(getByTestId('utility-icon')).toBeInTheDocument()
    expect(getByTestId('control-panel')).toBeInTheDocument()
  })

  it('should render without children', () => {
    const { container } = render(
      <RuioWrapper>
        <> </>
      </RuioWrapper>,
    )
    expect(container).toBeInTheDocument()
  })

  it('should render correctly with invalid children types', () => {
    const { container } = render(<RuioWrapper>{null}</RuioWrapper>)
    expect(container).toBeInTheDocument()
  })
})
