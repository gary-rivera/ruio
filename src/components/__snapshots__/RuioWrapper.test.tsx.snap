// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuioWrapper Snapshot Tests should match the snapshot with default children 1`] = `
<DocumentFragment>
  <div
    class="ruio-exclude"
    data-testid="ruio-control-panel"
    style="position: fixed; bottom: 80px; right: 20px; z-index: 9999; background-color: white; padding: 10px; border: 1px solid black; display: none;"
  >
    <label
      class="ruio-exclude"
    >
      Depth:
      <input
        class="ruio-exclude"
        max="10"
        min="1"
        type="number"
        value="1"
      />
    </label>
    <div
      class="ruio-exclude"
    >
      <button
        class="ruio-exclude"
        disabled=""
      >
        Select Element
      </button>
      <button>
        Enable Borders
      </button>
    </div>
  </div>
  <div
    class="ruio-exclude"
    data-testid="ruio-toggle-icon"
    style="position: fixed; bottom: 20px; right: 20px; z-index: 9999; background-color: lightcoral; padding: 10px; cursor: pointer;"
  >
    üõ†Ô∏è
  </div>
  <div>
    Snapshot Child
  </div>
</DocumentFragment>
`;
